mage_ai/data_preparation/models/pipeline.py

```diff
@@ -1905,6 +2047,20 @@ def __check_cycle(block: Block):
                     f'Pipeline is invalid: duplicate blocks with uuid {uuid}')
             check_block_uuids.add(uuid)
 
+    # Searches for a file with the specified name in the provided directory and its subdirectories.
+    # Returns str, the full path to the found file, or None if the file is not found.
```

Comment:
let's follow the docstring format for these kinds of comments. example: https://github.com/mage-ai/mage-ai/blob/master/mage_ai/data_preparation/models/block/__init__.py#L1563-L1584